.maintain/common/lib.sh

```diff
@@ -115,3 +119,25 @@ has_runtime_changes() {
     return 1
   fi
 }
+
+# Given an origin repo & PR,and a repository, will check if the given PR has a
+# companion PR in the target repo
+get_companion() {
+  origin_repo="$1"
+  pr_num="$2"
+  companion_repo="$3"
+  pr_data_file="$(mktemp)"
+  set_github_token
+  github_header="Authorization: token ${GITHUB_TOKEN}"
+  # get the last reference to a pr in the target repo
+  curl -sSL -H "${github_header}" -o "${pr_data_file}" \
+    "${api_base}/$origin_repo/pulls/$pr_num"
+
+  pr_body="$(sed -n -r 's/^[[:space:]]+"body": (".*")[^"]+$/\1/p' "${pr_data_file}")"
+
+  echo "${pr_body}" | sed -n -r \
```

Comment:
Does sed account for newlines in the JSON string?

This probably never happens but you might have a comment like this:

```
Something something companion

Bla bla another pull request: https://github.com/foo/pull/1
```

As I understand it, since you're not matching `[Cc]ompanion: [regex]` all in the same line, that comment would be detected although it should not be.

Using `echo -e` will output the JSON newlines as actual newlines so you'll be able to match it line-by-line.